### PR TITLE
Adding pyDOE3

### PR DESCRIPTION
**Pull request by relf**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by relf** [comment]
@conda-forge/help-python, ready for review, thanks!

File: recipes/pydoe3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+build:
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.8
+    - hatchling
```
Comment (ocefpaf) [inline recipes/pydoe3/meta.yaml:22]: Metadata has `hatch-vcs`. I don't know the first thing about hatch but maybe it is needed to add the version properly?

```suggestion
    - hatchling
    - hatch-vcs
```

File: recipes/pydoe3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  imports:
+    - pyDOE3
+
+about:
+  home: https://github.com/relf/pyDOE3
+  summary: 'Simple, fast, extensible JSON encoder/decoder for Python'
+  description: Fork of pyDOE2, package designed to help the scientist, engineer, statistician, etc., to construct appropriate experimental designs.
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (ocefpaf) [inline recipes/pydoe3/meta.yaml:39]:
```suggestion
  license_file: LICENSE
```

File: recipes/pydoe3/LICENSE [hunk 3]
```diff
@@ -0,0 +1,30 @@
+Copyright (c) 2014, Abraham D. Lee
```
Comment (ocefpaf) [inline recipes/pydoe3/LICENSE:1]: The license is in the tarball. You can remove this file.

Comment (relf) [reply recipes/pydoe3/LICENSE]: ok

**Pull request merged**